extends Control

@onready var shop_area: Area2D = $"../../Interactables/Shop"
@onready var shop_modal: Control = $ShopModal
@onready var game_over: Control = $GameOverModal

@onready var npc: Area2D = $"../../Interactables/NPC_Girl"
@onready var dialog: Control = $DialogModal

func _ready() -> void:
	shop_area.shop_pressed.connect(_on_shop_pressed)

	shop_modal.yes_pressed.connect(_on_shop_yes)
	shop_modal.no_pressed.connect(_on_shop_no)
	npc.npc_pressed.connect(_on_npc_pressed)
	dialog.finished.connect(_on_dialog_finished)

func _on_shop_pressed() -> void:
	shop_modal.open()
func _on_npc_pressed() -> void:
	# ÑÑ‚Ð°Ð²Ð¸Ð¼ Ð¿Ð°ÑƒÐ·Ñƒ
	get_tree().paused = true

	# Ñ€ÐµÐ¿Ð»Ð¸ÐºÐ¸ (Ð»ÑŽÐ±Ñ‹Ðµ, Ñ Ð¿Ñ€Ð¸Ð´ÑƒÐ¼Ð°Ð» ÐºÐ¾Ñ€Ð¾Ñ‚ÐºÐ¸Ð¹ Ð´Ð¸Ð°Ð»Ð¾Ð³ Ñ Ð²Ð°Ñ€Ð¸Ð°Ð½Ñ‚Ð¾Ð¼ Ð¾Ñ‚Ð²ÐµÑ‚Ð° Ð² ÑÐ»ÐµÐ´ÑƒÑŽÑ‰ÐµÐ¼ ÑˆÐ°Ð³Ðµ)
	var lines: Array[String] = [
		"â€” ÐŸÑ€Ð¸Ð²Ñ–Ñ‚! Ð¢Ð¸ Ñ‚ÐµÐ¶ Ð½Ð° Ð¿Ð°Ñ€Ð¸ Ð¹Ð´ÐµÑˆ?",
		"â€” Ð¢Ð°Ðº... Ð°Ð»Ðµ Ñ‰Ð¾ÑÑŒ ÑÑŒÐ¾Ð³Ð¾Ð´Ð½Ñ– Ð»Ñ–Ð½ÑŒ.",
		"â€” Ð¯ÐºÑ‰Ð¾ Ð¿Ñ–Ð´ÐµÑˆ Ð·Ð°Ñ€Ð°Ð· â€” Ñ‰Ðµ Ð²ÑÑ‚Ð¸Ð³Ð½ÐµÑˆ. ÐÐµ Ð·Ð°Ð²Ð¸ÑÐ°Ð¹ Ñ‚ÑƒÑ‚ ðŸ™‚"
	]
	dialog.open_dialog("ÐžÐ»Ñ", lines)


func _on_dialog_finished() -> void:
	# Ð¿Ð¾Ð¼ÐµÑ‚Ð¸Ð»Ð¸, Ñ‡Ñ‚Ð¾ ÑƒÐ¶Ðµ Ð¿Ð¾ÐºÐ°Ð·Ñ‹Ð²Ð°Ð»Ð¸
	if npc.has_method("mark_shown"):
		npc.mark_shown()

	# ÑÐ½Ð¸Ð¼Ð°ÐµÐ¼ Ð¿Ð°ÑƒÐ·Ñƒ
	get_tree().paused = false
func _on_shop_no() -> void:
	shop_modal.close()

func _on_shop_yes() -> void:
	shop_modal.close()
	get_tree().paused = true
	game_over.show_game_over_1("Ð’Ð¸ Ð¿Ñ€Ð¾Ð³Ñ€Ð°Ð»Ð¸.")
